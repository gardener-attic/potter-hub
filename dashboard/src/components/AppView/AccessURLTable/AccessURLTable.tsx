import * as React from "react";

import ResourceRef from "shared/ResourceRef";
import LoadingWrapper, { LoaderType } from "../../../components/LoadingWrapper";
import { IK8sList, IKubeItem, IResource, IServiceSpec } from "../../../shared/types";
import isSomeResourceLoading from "../helpers";
import AccessURLItem from "./AccessURLItem";
import { GetURLItemFromIngress } from "./AccessURLItem/AccessURLIngressHelper";
import { GetURLItemFromService } from "./AccessURLItem/AccessURLServiceHelper";

interface IAccessURLTableProps {
  services: Array<IKubeItem<IResource>>;
  ingresses: Array<IKubeItem<IResource>>;
  ingressRefs: ResourceRef[];
  getResource: (r: ResourceRef) => void;
}

class AccessURLTable extends React.Component<IAccessURLTableProps> {
  public componentDidMount() {
    this.fetchIngresses();
  }

  public componentDidUpdate(prevProps: IAccessURLTableProps) {
    if (prevProps.ingressRefs.length !== this.props.ingressRefs.length) {
      this.fetchIngresses();
    }
  }

  public render() {
    const { ingresses, services } = this.props;
    if (isSomeResourceLoading(ingresses.concat(services))) {
      return <LoadingWrapper type={LoaderType.Placeholder} />;
    }
    if (!this.hasItems(services, ingresses)) {
      return null;
    }
    return (
      <React.Fragment>
        <h6>Access URLs</h6>
        {this.accessTableSection()}
      </React.Fragment>
    );
  }

  private publicServices(): Array<IKubeItem<IResource>> {
    const { services } = this.props;
    const publicServices: Array<IKubeItem<IResource>> = [];
    services.forEach(s => {
      if (s.item) {
        const spec = s.item.spec as IServiceSpec;
        if (spec.type === "LoadBalancer") {
          publicServices.push(s);
        }
      }
    });
    return publicServices;
  }

  private accessTableSection() {
    const { ingresses } = this.props;
    let accessTableSection = <p>The current application does not expose a public URL.</p>;
    const publicServices = this.publicServices();
    if (publicServices.length > 0 || ingresses.length > 0) {
      accessTableSection = (
        <React.Fragment>
          <table>
            <thead>
              <tr>
                <th>RESOURCE</th>
                <th>TYPE</th>
                <th>URL</th>
              </tr>
            </thead>
            <tbody>
              {ingresses.map(i => this.renderTableEntry(i))}
              {publicServices.map(s => this.renderTableEntry(s))}
            </tbody>
          </table>
        </React.Fragment>
      );
    }
    return accessTableSection;
  }

  private renderTableEntry(i: IKubeItem<IResource>) {
    if (i.error) {
      return (
        <tr key={i.error.message}>
          <td colSpan={3}>Error: {i.error.message}</td>
        </tr>
      );
    }
    if (i.item) {
      const urlItem =
        i.item.kind === "Ingress" ? GetURLItemFromIngress(i.item) : GetURLItemFromService(i.item);
      return <AccessURLItem key={`accessURL/${i.item.metadata.name}`} URLItem={urlItem} />;
    }
    return;
  }

  private fetchIngresses() {
    // Fetch all related Ingress resources. We don't need to fetch Services as
    // they are expected to be watched by the ServiceTable.
    this.props.ingressRefs.forEach(r => this.props.getResource(r));
  }

  private elemHasItems(i: IKubeItem<IResource | IK8sList<IResource, {}>>) {
    if (i.error) {
      return true;
    }
    if (i.item) {
      const list = i.item as IK8sList<IResource, {}>;
      if (list.items && list.items.length === 0) {
        return false;
      }
      return true;
    }
    return false;
  }

  private hasItems(
    svcs: Array<IKubeItem<IResource | IK8sList<IResource, {}>>>,
    ingresses: Array<IKubeItem<IResource | IK8sList<IResource, {}>>>,
  ) {
    return (
      (svcs.length && svcs.some(svc => this.elemHasItems(svc))) ||
      (ingresses.length && ingresses.some(ingress => this.elemHasItems(ingress)))
    );
  }
}

export default AccessURLTable;
