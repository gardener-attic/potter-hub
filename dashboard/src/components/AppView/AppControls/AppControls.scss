.upgrade-button {
  margin-right: 0.4em;
}

.notification {
  position: absolute;
  right: 15%;
  top: 22%;
}

.upgrade-text {
  padding-right: 1.25em;
  color: #008145;
}

.button:hover {
  .upgrade-text {
    color: inherit;
  }
  .notification {
    fill: transparent;
  }
}
