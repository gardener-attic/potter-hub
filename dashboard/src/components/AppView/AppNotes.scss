.Terminal {
  overflow-wrap: break-word;
}
