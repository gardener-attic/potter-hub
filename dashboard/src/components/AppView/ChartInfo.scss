.ChartInfo > a {
  display: block;
  width: 100%;
}

.ChartInfoUpdate {
  color: white;
  width: 100%;
  margin-bottom: 0.5em;
  text-align: center;
  padding: 0.25em 0;
  font-size: 14px;
  background: #008145;
}

.ChartControllerManaged {
  text-align: center;
  background: rgba(9, 4, 80, 0.68);
  border-radius: 10px;
}
