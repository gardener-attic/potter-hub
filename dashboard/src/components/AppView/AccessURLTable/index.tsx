import AccessURLTable from "./AccessURLTable";

export default AccessURLTable;
