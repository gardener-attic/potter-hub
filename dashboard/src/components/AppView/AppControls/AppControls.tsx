import { RouterAction } from "connected-react-router";
import * as React from "react";
import { Redirect } from "react-router";

import { IRelease } from "shared/types";
import RollbackButtonContainer from "../../../containers/RollbackButtonContainer";
import ConfirmDialog from "../../ConfirmDialog";
import LoadingWrapper from "../../LoadingWrapper";
import "./AppControls.css";
import UpgradeButton from "./UpgradeButton";

interface IAppControlsProps {
  app: IRelease;
  deleteApp: (keepHistory: boolean) => Promise<boolean>;
  push: (location: string) => RouterAction;
}

interface IAppControlsState {
  migrate: boolean;
  modalIsOpen: boolean;
  redirectToAppList: boolean;
  upgrade: boolean;
  deleting: boolean;
  keepHistory: boolean;
}

class AppControls extends React.Component<IAppControlsProps, IAppControlsState> {
  public state: IAppControlsState = {
    deleting: false,
    migrate: false,
    modalIsOpen: false,
    keepHistory: false,
    redirectToAppList: false,
    upgrade: false,
  };

  public render() {
    const { app, push } = this.props;
    const { name, namespace } = app;
    const deleted = app.info && app.info.deleted;
    if (!name || !namespace) {
      return <LoadingWrapper />;
    }
    return (
      <div className="AppControls">
        {/* If the app has been deleted hide the upgrade button */}
        {!deleted && (
          <React.Fragment>
            <UpgradeButton
              newVersion={app.updateInfo && !app.updateInfo.upToDate}
              releaseName={name}
              releaseNamespace={namespace}
              push={push}
            />
            {/* We only show the rollback button if there are versions to rollback */}
            {app.version > 1 && (
              <RollbackButtonContainer
                releaseName={name}
                namespace={namespace}
                revision={app.version}
              />
            )}
          </React.Fragment>
        )}
        <button className="button button-danger" onClick={this.openModel}>
          {deleted ? "Purge" : "Delete"}
        </button>
        <ConfirmDialog
          onConfirm={this.handleDeleteClick}
          modalIsOpen={this.state.modalIsOpen}
          loading={this.state.deleting}
          closeModal={this.closeModal}
          extraElem={
            deleted ? (
              undefined
            ) : (
              <div className="margin-b-normal text-c">
                <label className="checkbox margin-r-big">
                  <input type="checkbox" onChange={this.togglePurge} />
                  <span>Keep History</span>
                </label>
              </div>
            )
          }
        />
        {this.state.redirectToAppList && <Redirect to={`/apps/ns/${namespace}`} />}
      </div>
    );
  }

  public openModel = () => {
    this.setState({
      modalIsOpen: true,
    });
  };

  public closeModal = async () => {
    this.setState({
      modalIsOpen: false,
    });
  };

  public handleDeleteClick = async () => {
    this.setState({ deleting: true });
    // Purge the release if the application has been already deleted
    const alreadyDeleted = this.props.app.info && !!this.props.app.info.deleted;
    const deleted = await this.props.deleteApp(!alreadyDeleted && this.state.keepHistory);
    const s: Partial<IAppControlsState> = { modalIsOpen: false };
    if (deleted) {
      s.redirectToAppList = true;
    }
    this.setState(s as IAppControlsState);
  };

  private togglePurge = () => {
    this.setState({ keepHistory: !this.state.keepHistory });
  };
}

export default AppControls;
