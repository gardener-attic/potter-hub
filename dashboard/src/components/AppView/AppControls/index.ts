import AppControls from "./AppControls";

export default AppControls;
