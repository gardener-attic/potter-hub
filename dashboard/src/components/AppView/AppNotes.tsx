import * as React from "react";

import "./AppNotes.css";

interface IAppNotesProps {
  notes?: string | null;
}

class AppNotes extends React.PureComponent<IAppNotesProps> {
  public render() {
    const { notes } = this.props;
    return notes ? (
      <React.Fragment>
        <h6>Notes</h6>
        <section className="AppNotes Terminal elevation-1">
          <div className="Terminal__Top type-small">
            <div className="Terminal__Top__Buttons">
              <span className="Terminal__Top__Button Terminal__Top__Button--red" />
              <span className="Terminal__Top__Button Terminal__Top__Button--yellow" />
              <span className="Terminal__Top__Button Terminal__Top__Button--green" />
            </div>
          </div>
          <div className="Terminal__Tab">
            <pre className="Terminal__Code">
              <code>{notes}</code>
            </pre>
          </div>
        </section>
      </React.Fragment>
    ) : (
      ""
    );
  }
}

export default AppNotes;
