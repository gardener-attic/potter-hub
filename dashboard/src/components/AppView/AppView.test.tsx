import { shallow } from "enzyme";
import context from "jest-plugin-context";
import { safeDump as yamlSafeDump, YAMLException } from "js-yaml";
import * as React from "react";

import AccessURLTable from "../../containers/AccessURLTableContainer";
import ApplicationStatus from "../../containers/ApplicationStatusContainer";
import ApplicationStatusContainer from "../../containers/ApplicationStatusContainer";
import { hapi } from "../../shared/hapi/release";
import ResourceRef from "../../shared/ResourceRef";
import itBehavesLike from "../../shared/specs";
import { ForbiddenError, IResource, NotFoundError } from "../../shared/types";
import { ErrorSelector } from "../ErrorAlert";
import PermissionsErrorPage from "../ErrorAlert/PermissionsErrorAlert";
import AppControls from "./AppControls";
import AppNotes from "./AppNotes";
import AppViewComponent, { IAppViewProps } from "./AppView";
import ChartInfo from "./ChartInfo";
import ResourceTable from "./ResourceTable";

describe("AppViewComponent", () => {
  // Generates a Yaml file separated by --- containing every object passed.
  const generateYamlManifest = (items: any[]): string => {
    let yamlManifest = "";
    items.forEach(i => {
      yamlManifest += "---\n" + yamlSafeDump(i);
    });
    return yamlManifest;
  };

  const appRelease = hapi.release.Release.create({
    info: hapi.release.Info.create(),
    namespace: "weee",
  });

  const validProps: IAppViewProps = {
    app: appRelease,
    deleteApp: jest.fn(),
    deleteError: undefined,
    error: undefined,
    getAppWithUpdateInfo: jest.fn(),
    namespace: "my-happy-place",
    releaseName: "mr-sunshine",
    push: jest.fn(),
  };

  const resources = {
    configMap: { apiVersion: "v1", kind: "ConfigMap", metadata: { name: "cm-one" } },
    deployment: {
      apiVersion: "apps/v1beta1",
      kind: "Deployment",
      metadata: { name: "deployment-one" },
    } as IResource,
    service: { apiVersion: "v1", kind: "Service", metadata: { name: "svc-one" } } as IResource,
    ingress: {
      apiVersion: "extensions/v1beta1",
      kind: "Ingress",
      metadata: { name: "ingress-one" },
    } as IResource,
    secret: {
      apiVersion: "v1",
      kind: "Secret",
      metadata: { name: "secret-one" },
      type: "Opaque",
    } as IResource,
    daemonset: {
      apiVersion: "apps/v1beta1",
      kind: "DaemonSet",
      metadata: { name: "daemonset-one" },
    } as IResource,
    statefulset: {
      apiVersion: "apps/v1beta1",
      kind: "StatefulSet",
      metadata: { name: "statefulset-one" },
    } as IResource,
  };

  context("when app info is null", () => {
    itBehavesLike("aLoadingComponent", {
      component: AppViewComponent,
      props: {
        ...validProps,
        app: { ...validProps.app, info: null },
      },
    });
  });

  describe("State initialization", () => {
    /*
      The imported manifest contains one deployment, one service, one config map and some bogus manifests.
    */
    it("sets ResourceRefs for its deployments, services, ingresses and secrets", () => {
      const manifest = generateYamlManifest([
        resources.deployment,
        resources.service,
        resources.configMap,
        resources.ingress,
        resources.secret,
      ]);

      const wrapper = shallow(<AppViewComponent {...validProps} />);
      validProps.app.manifest = manifest;
      // setProps again so we trigger componentWillReceiveProps
      wrapper.setProps(validProps);

      expect(wrapper.state("deployRefs")).toEqual([
        new ResourceRef(resources.deployment, appRelease.namespace),
      ]);
      expect(wrapper.state("serviceRefs")).toEqual([
        new ResourceRef(resources.service, appRelease.namespace),
      ]);
      expect(wrapper.state("ingressRefs")).toEqual([
        new ResourceRef(resources.ingress, appRelease.namespace),
      ]);
      expect(wrapper.state("secretRefs")).toEqual([
        new ResourceRef(resources.secret, appRelease.namespace),
      ]);
    });

    it("stores other k8s resources directly in the state", () => {
      const wrapper = shallow(<AppViewComponent {...validProps} />);
      const manifest = generateYamlManifest([
        resources.deployment,
        resources.service,
        resources.configMap,
        resources.secret,
      ]);

      validProps.app.manifest = manifest;
      wrapper.setProps(validProps);

      const otherResources: ResourceRef[] = wrapper.state("otherResources");
      const configMap = otherResources[0];
      // It should skip deployments, services and secrets from "other resources"
      expect(otherResources.length).toEqual(1);

      expect(configMap).toBeDefined();
      expect(configMap.name).toEqual("cm-one");
    });

    it("does not store empty resources, bogus or without kind attribute", () => {
      const wrapper = shallow(<AppViewComponent {...validProps} />);
      const manifest = generateYamlManifest([
        { apiVersion: "v1", metadata: { name: "cm-one" } },
        {},
        "# This is a comment",
        " ",
      ]);

      validProps.app.manifest = manifest;
      wrapper.setProps(validProps);

      expect(wrapper.state("otherResources")).toEqual([]);
      expect(wrapper.state("deployRefs")).toEqual([]);
      expect(wrapper.state("serviceRefs")).toEqual([]);
      expect(wrapper.state("ingressRefs")).toEqual([]);
      expect(wrapper.state("secretRefs")).toEqual([]);
    });

    // See https://github.com/kubeapps/kubeapps/issues/632
    it("supports manifests with duplicated keys", () => {
      const wrapper = shallow(<AppViewComponent {...validProps} />);
      const manifest = `
      apiVersion: v1
      metadata:
        name: cm-one
        labels:
          chart: cm-1.2.3
          chart: cm-1.2.3
`;

      validProps.app.manifest = manifest;
      expect(() => {
        wrapper.setProps(validProps);
      }).not.toThrow(YAMLException);
    });
  });

  describe("renderization", () => {
    it("renders all the elements of an application", () => {
      const wrapper = shallow(<AppViewComponent {...validProps} />);
      expect(wrapper.find(ChartInfo).exists()).toBe(true);
      expect(wrapper.find(ApplicationStatus).exists()).toBe(true);
      expect(wrapper.find(AppControls).exists()).toBe(true);
      expect(wrapper.find(AppNotes).exists()).toBe(true);
      expect(wrapper.find(ResourceTable).exists()).toBe(true);
      expect(wrapper.find(AccessURLTable).exists()).toBe(true);
    });

    it("renders an error if error prop is set", () => {
      const wrapper = shallow(<AppViewComponent {...validProps} error={new NotFoundError()} />);
      const err = wrapper.find(ErrorSelector);
      expect(err.exists()).toBe(true);
      expect(err.html()).toContain("Application mr-sunshine not found");
    });

    it("renders a forbidden delete-error if if the deleteError prop is a ForbiddenError", () => {
      const wrapper = shallow(
        <AppViewComponent {...validProps} deleteError={new ForbiddenError()} />,
      );
      const err = wrapper.find(ErrorSelector);
      expect(err.exists()).toBe(true);
      expect(
        err
          .shallow()
          .find(PermissionsErrorPage)
          .props(),
      ).toMatchObject({
        action: "delete Application mr-sunshine",
        namespace: "my-happy-place",
      });
    });

    it("renders a not-found delete-error if the deleteError prop is a NotFound error", () => {
      const wrapper = shallow(
        <AppViewComponent {...validProps} deleteError={new NotFoundError()} />,
      );
      const err = wrapper.find(ErrorSelector);
      expect(err.exists()).toBe(true);
      expect(err.html()).toContain("Application mr-sunshine not found");
    });
  });

  it("forwards services/ingresses", () => {
    const wrapper = shallow(<AppViewComponent {...validProps} />);
    const ingress = {
      isFetching: false,
      item: {
        metadata: {
          name: "foo",
        },
        spec: {
          rules: [
            {
              host: "foo.bar",
              http: {
                paths: [{ path: "/ready" }],
              },
            },
          ],
        },
      },
    };
    const ingressRefs = [new ResourceRef(ingress.item as IResource, "default")];
    const service = {
      isFetching: false,
      item: {
        apiVersion: "v1",
        kind: "Service",
        metadata: {
          name: "foo",
        },
        spec: {},
      },
    };
    const serviceRefs = [new ResourceRef(service.item as IResource, "default")];

    wrapper.setState({ ingressRefs, serviceRefs });

    const accessURLTable = wrapper.find(AccessURLTable);
    expect(accessURLTable).toExist();
    expect(accessURLTable.props()).toMatchObject({ ingressRefs, serviceRefs });

    const svcTable = wrapper.find(ResourceTable).findWhere(t => t.prop("title") === "Services");
    expect(svcTable).toExist();
    expect(svcTable.prop("resourceRefs")).toEqual(serviceRefs);
  });

  it("forwards deployments", () => {
    const wrapper = shallow(<AppViewComponent {...validProps} />);
    const deployment = {
      metadata: {
        name: "foo",
      },
      spec: {},
    };
    const deployRefs = [new ResourceRef(deployment as IResource, "default")];

    wrapper.setState({ deployRefs });

    const depTable = wrapper
      .find(ResourceTable)
      .filterWhere(e => e.prop("title") === "Deployments");
    expect(depTable).toExist();
    expect(depTable.prop("resourceRefs")).toEqual(deployRefs);
  });

  it("forwards statefulsets", () => {
    const wrapper = shallow(<AppViewComponent {...validProps} />);
    const r = {
      metadata: {
        name: "foo",
      },
      spec: {},
    };
    const ref = [new ResourceRef(r as IResource, "default")];

    wrapper.setState({ statefulSetRefs: ref });

    const depTable = wrapper
      .find(ResourceTable)
      .filterWhere(e => e.prop("title") === "StatefulSets");
    expect(depTable).toExist();
    expect(depTable.prop("resourceRefs")).toEqual(ref);
  });

  it("forwards daemonsets", () => {
    const wrapper = shallow(<AppViewComponent {...validProps} />);
    const r = {
      metadata: {
        name: "foo",
      },
      spec: {},
    };
    const ref = [new ResourceRef(r as IResource, "default")];

    wrapper.setState({ daemonSetRefs: ref });

    const depTable = wrapper.find(ResourceTable).filterWhere(e => e.prop("title") === "DaemonSets");
    expect(depTable).toExist();
    expect(depTable.prop("resourceRefs")).toEqual(ref);
  });

  it("forwards other resources", () => {
    const wrapper = shallow(<AppViewComponent {...validProps} />);
    const otherResource = {
      metadata: {
        name: "foo",
      },
      spec: {},
    };
    const otherResources = [otherResource];

    wrapper.setState({ otherResources });

    const orTable = wrapper
      .find(ResourceTable)
      .filterWhere(e => e.prop("title") === "Other Resources");
    expect(orTable).toExist();
    expect(orTable.prop("resourceRefs")).toEqual([otherResource]);
  });

  it("renders a list of resources", () => {
    const obj = { kind: "ClusterRole", metadata: { name: "foo" } } as IResource;
    const list = {
      kind: "List",
      items: [obj, resources.deployment],
    };
    const manifest = generateYamlManifest([resources.service, list]);

    const wrapper = shallow(<AppViewComponent {...validProps} />);
    validProps.app.manifest = manifest;
    // setProps again so we trigger componentWillReceiveProps
    wrapper.setProps(validProps);

    expect(wrapper.state()).toMatchObject({
      deployRefs: [new ResourceRef(resources.deployment, appRelease.namespace)],
      serviceRefs: [new ResourceRef(resources.service, appRelease.namespace)],
      otherResources: [new ResourceRef(obj, appRelease.namespace)],
    });
  });

  it("renders a list of roles", () => {
    const obj = { kind: "ClusterRole", metadata: { name: "foo" } } as IResource;
    const list = {
      kind: "RoleList",
      items: [obj, resources.deployment],
    };
    const manifest = generateYamlManifest([resources.service, list]);

    const wrapper = shallow(<AppViewComponent {...validProps} />);
    validProps.app.manifest = manifest;
    // setProps again so we trigger componentWillReceiveProps
    wrapper.setProps(validProps);

    expect(wrapper.state()).toMatchObject({
      deployRefs: [new ResourceRef(resources.deployment, appRelease.namespace)],
      serviceRefs: [new ResourceRef(resources.service, appRelease.namespace)],
      otherResources: [new ResourceRef(obj, appRelease.namespace)],
    });
  });

  it("forwards statefulsets and daemonsets", () => {
    const r = [resources.statefulset, resources.daemonset];
    const manifest = generateYamlManifest(r);
    const wrapper = shallow(<AppViewComponent {...validProps} />);
    validProps.app.manifest = manifest;
    // setProps again so we trigger componentWillReceiveProps
    wrapper.setProps(validProps);

    const applicationStatus = wrapper.find(ApplicationStatusContainer);
    expect(applicationStatus).toExist();

    expect(applicationStatus.prop("statefulsetRefs")).toEqual([
      new ResourceRef(resources.statefulset, appRelease.namespace),
    ]);
    expect(applicationStatus.prop("daemonsetRefs")).toEqual([
      new ResourceRef(resources.daemonset, appRelease.namespace),
    ]);
  });
});
