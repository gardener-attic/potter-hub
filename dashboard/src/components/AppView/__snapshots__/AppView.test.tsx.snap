// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppViewComponent when app info is null loading spinner matches the snapshot 1`] = `
<LoadingWrapper
  loaded={false}
  type={0}
/>
`;
